blog/2025/passkeys-support-26-4.adoc

```diff
@@ -0,0 +1,105 @@
+
+Passkeys have been available in Keycloak since version 23.0.0 as a preview feature. We are happy to announce official support for passkeys in upcoming **Keycloak 26.4.0**.
+
+== What is passkey?
+
+Definition from https://fidoalliance.org[FIDO Alliance]
+====
+A passkey is a FIDO authentication credential based on FIDO standards, that allows a user to sign in to apps and websites with the same process that they use to unlock their device (biometrics, PIN, or pattern).
+Passkeys are FIDO cryptographic credentials that are tied to a user’s account on a website or application.
+====
+
+https://www.passkeycentral.org/home[More info]
+
+== What's new?
+
+Passkeys are now seamlessly integrated to Keycloak using both conditional and modal UI. There is no need to modify default browser flow to use passkeys. Passkeys support is not enabled aby default. It needs to be enabled in the *WebAuthn Passwordless Policy* (*Authentication* → *Policies* → *Webauthn Passwordless Policy*).
+
+There is new *Conditional - credential* authenticator that checks if a specific credential type (passkey) has been used during
+the authentication process. It is added to the default *browser* flow to skip 2FA in case a passkey was used to log in as
+the primary credential.
+
+Further quite hidden passkey support is also in the re-authentication form, where users can choose passkey as well as password.
+
+For more information check https://www.keycloak.org/docs/latest/server_admin/index.html#_passkeys-conditional-ui[Keycloak Server Administration Guide].
+
+== A few examples
+
+Let's start form the fresh Keycloak database. The first steps are obvious ones.
+
+- create admin user
+- create one test user
+- enable passkey support in (*Authentication* → *Policies* → *Webauthn Passwordless Policy*)
+- set default required action in (*Authentication* → *Required actions* → *Webauthn Register Passwordless*)
+
+No need for any other changes.
+
+=== Conditional UI
+
+Conditional UI is displayed when other components of the platform supports this UI stile.
+It can be achieved for example on Linux using Google Chrome browser and 1password password manager with Chrome extension for 1password.
+
+Registering passkey stored in 1password manager:
+--
+++++
+<div class="paragraph">
+</style>
+<img src="${blogImages}/passkeys/1pwd-register.png" alt="Registerering 1password passkey" style="width: 100%; max-width: 863px; object-fit: cover; object-fit: none; object-position: 0 0">
+</div>
+++++
+--
+Using passkey stored in 1password manager:
+--
+++++
+<div class="paragraph">
+</style>
+<img src="${blogImages}/passkeys/1pwd-use.png" alt="Using 1password passkey" style="width: 100%; max-width: 863px; object-fit: cover; object-fit: none; object-position: 0 0">
+</div>
+++++
+--
+
+=== Modal UI
+
+Modal UI is an older style and it is used mainly with hardware keys asking for PIN or biometric (such as Yubikeys). It is directly supported in all major browsers.
```

Comment:
The same, we can add here that the modal UI is presented as a separate button all the times that conditional UI is used (username input is presented) and also for re-authentication (when the user is already selected but password is required).